Hailstone.java

```diff
@@ -10,7 +10,23 @@
 
 public class Hailstone extends ConsoleProgram {
 	public void run() {
+		int x = readInt ("Enter a number;");
+		//
+		while (x > 1){
+			if (x % 2 != 0){
+				println ("+ x is odd");
+				int y = (3*x) + 1;
```

Comment:
Here you are supposed to update the value of x to 3x + 1 but instead you assigned 3x + 1 to y. Because of this your while loop will never stop since the value of x never changes. Your code should be:
`x = (3*x) + 1;` instead of 
`int y = (3*x) + 1;`